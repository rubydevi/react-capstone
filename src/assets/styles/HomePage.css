.input-container {
  position: relative;
  padding: 30% 5%;
  width: 100%;
  display: flex;
  justify-content: center;

  /* set background image */
  background-image: url('../images/homeBackground.jpg');
  background-size: cover;
  background-repeat: no-repeat;
}

.search-icon {
  position: absolute;
  top: 50%;
  left: 75%;
  transform: translateY(-50%);
}

.search {
  color: grey;
}

input[type='text'] {
  width: 70%;
  border-radius: 15px;
}

.pokemon-grid {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  list-style: none;
  padding: 0;
}

.pokemon-grid-item {
  display: flex;
  align-items: flex-end;
  flex-direction: column;
  padding: 10%;
  justify-content: flex-end;
  background-color: #f65090;
  color: #fffeff;
  row-gap: 57px;
}

.pokemon-grid-item:nth-child(4n - 3),
.pokemon-grid-item:nth-child(4n - 4) {
  background-color: #cf4278;
}

.pokemon-grid-item:nth-child(4n - 2),
.pokemon-grid-item:nth-child(4n - 1) {
  background-color: #f65090;
}
