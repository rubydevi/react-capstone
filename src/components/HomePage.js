import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { FaSearch, FaArrowCircleRight } from 'react-icons/fa';
import { fetchPokemons, selectPokemon } from '../redux/pokemon/pokemonSlice';
import '../assets/styles/HomePage.css';

const HomePage = () => {
  const dispatch = useDispatch();
  const pokemons = useSelector((state) => state.pokemons.pokemons);
  const [searchTerm, setSearchTerm] = useState('');

  useEffect(() => {
    if (pokemons.length === 0) {
      dispatch(fetchPokemons());
    }
  }, [dispatch, pokemons]);

  const handlePokemonClick = (pokemon) => {
    dispatch(selectPokemon(pokemon));
  };

  const filterPokemons = (pokemon) => pokemon.name.toLowerCase().includes(searchTerm.toLowerCase());

  const filteredPokemons = pokemons.filter(filterPokemons);

  return (
    <div>
      <div className="input-container">
        <span className="search-icon">
          <FaSearch className="search" />
        </span>
        <input
          type="text"
          placeholder="Search by name"
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
        />
      </div>
      <ul className="pokemon-grid">
        {filteredPokemons.map((pokemon) => (
          <li key={pokemon.id} className="pokemon-grid-item">
            <FaArrowCircleRight />
            <Link to={`/details/${pokemon.name}`} onClick={() => handlePokemonClick(pokemon)}>
              <p>
                {(pokemon.name).toUpperCase()}
              </p>
              <p>
                Number:
                {pokemon.value}
              </p>
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default HomePage;
