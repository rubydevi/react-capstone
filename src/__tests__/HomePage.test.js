import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import HomePage from '../components/HomePage';

const mockStore = configureStore([]);

test('HomePage component snapshot', () => {
  const initialState = {
    pokemons: {
      pokemons: [
        { id: 1, name: 'Pikachu', value: 25 },
        { id: 2, name: 'Charmander', value: 4 },
      ],
    },
  };
  const store = mockStore(initialState);

  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <HomePage />
        </Router>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
