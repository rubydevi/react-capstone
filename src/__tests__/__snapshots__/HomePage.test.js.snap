// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component snapshot 1`] = `
<div>
  <div
    className="input-container"
  >
    <span
      className="search-icon"
    >
      <svg
        className="search"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </span>
    <input
      onChange={[Function]}
      placeholder="Search by name"
      type="text"
      value=""
    />
  </div>
  <ul
    className="pokemon-grid"
  >
    <li
      className="pokemon-grid-item"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
        />
      </svg>
      <a
        href="/details/Pikachu"
        onClick={[Function]}
      >
        <p>
          PIKACHU
        </p>
        <p>
          Number:
          25
        </p>
      </a>
    </li>
    <li
      className="pokemon-grid-item"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
        />
      </svg>
      <a
        href="/details/Charmander"
        onClick={[Function]}
      >
        <p>
          CHARMANDER
        </p>
        <p>
          Number:
          4
        </p>
      </a>
    </li>
  </ul>
</div>
`;
